### PR TITLE
Api 37728 claims controller refactor

### PR DESCRIPTION
**Pull request by rockwellwindsor-va**
**Note**: Delete the description statements, complete each step. **None are optional**, but can be justified as to why they cannot be completed as written. Provide known gaps to testing that may raise the risk of merging to production.

## Summary

- *This work is behind a feature toggle (flipper): YES/NO*
- *(Summarize the changes that have been made to the platform)*
- *(If bug, how to reproduce)*
- *(What is the solution, why is this the solution?)*
- *(Which team do you work for, does your team own the maintenance of this component?)*
- *(If introducing a flipper, what is the success criteria being targeted?)*

## Related issue(s)

- *Link to ticket created in va.gov-team repo OR screenshot of Jira ticket if your team uses Jira*
- *Link to previous change of the code/bug (if applicable)*
- *Link to epic if not included in ticket*

## Testing done

- [ ] *New code is covered by unit tests*
- *Describe what the old behavior was prior to the change*
- *Describe the steps required to verify your changes are working as expected. Exclusively stating 'Specs run' is NOT acceptable as appropriate testing*
- *If this work is behind a flipper:*
  - *Tests need to be written for both the flipper on and flipper off scenarios. [Docs](https://depo-platform-documentation.scrollhelp.site/developer-docs/feature-toggles-guide#Featuretogglesguide-Backendexample).*
  - *What is the testing plan for rolling out the feature?*

## Screenshots
_Note: Optional_

## What areas of the site does it impact?
*(Describe what parts of the site are impacted and*if*code touched other areas)*

## Acceptance criteria

- [ ]  I fixed|updated|added unit tests and integration tests for each feature (if applicable).
- [ ]  No error nor warning in the console.
- [ ]  Events are being sent to the appropriate logging solution
- [ ]  Documentation has been updated (link to documentation)
- [ ]  No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs
- [ ]  Feature/bug has a monitor built into Datadog (if applicable)
- [ ]  If app impacted requires authentication, did you login to a local build and verify all authenticated routes work as expected
- [ ]  I added a screenshot of the developed feature

## Requested Feedback

(OPTIONAL)_What should the reviewers know in addition to the above. Is there anything specific you wish the reviewer to assist with. Do you have any concerns with this PR, why?_

### CODE REVIEW

File: modules/claims_api/spec/factories/auto_established_claims.rb [hunk 1]
```diff
@@ -241,7 +241,7 @@
     phase_chngd_dt { Faker::Time.backward(days: 5, period: :morning) }
     phase_type { 'Pending Decision Approval' }
     phase_type_change_ind { '76' }
-    ptcpnt_vet_id { Faker::Number.number(digits: 9) }
+    ptcpnt_vet_id { Faker::Number.number(digits: 9).to_s }
```
Comment (rockwellwindsor-va) [inline modules/claims_api/spec/factories/auto_established_claims.rb:244]: This was just a mistake on my part way back.  This should be a string.  The test still returned green, but not for the reason it should have.

File: modules/claims_api/spec/requests/v2/veterans/claims_request_spec.rb [hunk 2]
```diff
@@ -548,8 +548,8 @@
       end
     end
 
-    context 'show with validate_id_with_icn when there is a claimant ID in place of the verteran ID' do
-      describe ' BGS attributes (w/ Claimant ID replacing vet ID)' do
+    context 'show with validate_id_with_icn when there is a claimant ID in place of the veteran ID' do
+      describe 'BGS attributes (w/ Claimant ID replacing vet ID)' do
```
Comment (rockwellwindsor-va) [inline modules/claims_api/spec/requests/v2/veterans/claims_request_spec.rb:552]: Typo fix

File: modules/claims_api/app/controllers/claims_api/v2/veterans/claims_controller.rb [hunk 3]
```diff
@@ -2,146 +2,88 @@
...
-                        end
-
-          if clm_prtcpnt_cannot_access_claim?(clm_prtcpnt_vet_id, clm_prtcpnt_clmnt_id) && veteran_icn != request_icn
-            raise ::Common::Exceptions::ResourceNotFound.new(
-              detail: 'Invalid claim ID for the veteran identified.'
-            )
-          end
+        # Does not work exactly as it appears
+        def render_if_no_claims
+          render json: [] && return unless @bgs_claims || @lighthouse_claims
```
Comment (stiehlrod) [inline modules/claims_api/app/controllers/claims_api/v2/veterans/claims_controller.rb:36]: @rockwellwindsor-va What actually happens?

Comment (rockwellwindsor-va) [reply modules/claims_api/app/controllers/claims_api/v2/veterans/claims_controller.rb:36]: @stiehlrod It just returns empty
Screenshot 2024-08-09 at 4 37 44 PM

**Pull request closed without merging**